pymatgen/core/composition.py

```diff
@@ -317,7 +317,7 @@ def is_element(self) -> bool:
 
     def copy(self) -> Self:
         """A copy of the composition."""
-        return type(self)(self, allow_negative=self.allow_negative)
+        return Composition(self, allow_negative=self.allow_negative)
```

Comment:
Need some help on this, replacing this seems to break tests for `core.ion`, I assume there might be some issue with the inheritance of `Ion`?

Reply:
i assume you mean this error?

```py
E       TypeError: Ion.__init__() got an unexpected keyword argument 'allow_negative'
```

should be fixed by just forwarding any additional keywords in the `Ion.__init__` to `Composition.__init__` (see 7639b91)

Reply:
Yes, and thanks for the fix!